### DISCUSSION TITLE
Support task list selection action

### DISCUSSION THREAD

**Discussion by moophis** [gonzalezreal/swift-markdown-ui#204] [Ideas]
It looks like the task bullets are not selectable. Ideally, we can make each list tappable to select/unselect items and provide a callback to persist underlying content changes. Something like:
```
Markdown(text)
  .onTextChange { newText in // when a task item is tapped
    text = newText
  }
```
I would appreciate it if we're able to support this functionality. Thanks!

**Comment by gonzalezreal** [comment]
Hi @moophis,

The way to deal with item selection in SwiftUI is typically by providing a binding to some external state that the view will update as the user selects or deselects items. However, since a Markdown text can contain many task lists, developing an API that makes sense will take some time. That is, clients should provide a way to identify the task list, apart from the binding to the selection state.

Any suggestions or ideas for this API are welcome!

**Comment by moophis** [comment]
Thanks for the reply! 

Maybe a quick but less efficient idea is probably to modify the text string data directly or provide a new copy to the caller (e.g., replace `- [ ] a task item` with `- [ x ] a task item`). To avoid having to parse and re-render the entire markdown text, we could probably maintain an internal state in `TaskListItemView` and change the selection status based on that state once it's ever toggled (otherwise we stick to the text data itself). I'm not familiar with the internal workings (maybe I need to study more) but hopefully, that will work.
